tests/ui/lint/reference_casting.rs

```diff
@@ -144,6 +145,7 @@ unsafe fn no_warn() {
     let mut value = 3;
     let value: *const i32 = &mut value;
     *(value as *const i16 as *mut i16) = 42;
+    *RAW_PTR = 42;
```

Comment:
Could you add a comment explaining why this line is added?

Reply:
Sure, done.